dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaCMFHostsTaskTest.java

```diff
@@ -177,29 +135,65 @@ public void noHostsInClusterResponse_throwsException() throws Exception {
   }
 
   @Test
-  public void noClusterId_skip_writes() throws Exception {
-    handle.initClusters(ImmutableList.of(ClouderaClusterDTO.create(null, "single cluster")));
+  public void doRun_initClusterWithoutId_skipWrites() throws Exception {
+    initClusters(ClouderaClusterDTO.create(null, "single cluster"));
 
     task.doRun(context, sink, handle);
 
-    verify(httpClient, never()).execute(any());
-
     verifyNoWrites();
   }
 
   @Test
-  public void clustersWereNotInitialized_throwsException() throws Exception {
+  public void doRun_clustersWereNotInitialized_throwsCriticalException() throws Exception {
     assertNull(handle.getClusters());
 
     MetadataDumperUsageException exception =
         assertThrows(MetadataDumperUsageException.class, () -> task.doRun(context, sink, handle));
 
     assertEquals(
         "Cloudera clusters must be initialized before hosts dumping.", exception.getMessage());
+    verifyNoWrites();
+  }
 
+  @Test
+  public void doRun_clouderaReturnsInvalidJson_continueTaskWithoutWriting() throws Exception {
+    initClusters(ClouderaClusterDTO.create("id1", "first-cluster"));
+    mockCMFHostResponse("id1", "first-cluster", "[}");
     verifyNoWrites();
   }
 
+  private void initClusters(ClouderaClusterDTO... clusters) {
+    handle.initClusters(Arrays.asList(clusters));
+  }
+
+  private void mockCMFHostResponse(String clusterId, String clusterName, String jsonHosts)
```

Comment:
The name of the method is not represent what the method does. 
Maybe `stubCMFHostAPI` or something different, but not `mock`

Reply:
Got it, thanks! Renamed.